apps/storefront/src/pages/ShoppingListDetails/components/ShoppingDetailFooter.tsx

```diff
@@ -352,14 +344,18 @@ function ShoppingDetailFooter(props: ShoppingDetailFooterProps) {
       return;
     }
 
+    const addToCart = backendValidationEnabled ? handleAddToCartBackend : handleAddToCartOnFrontend;
+
     handleClose();
 
-    setLoading(true);
+    setValidateFailureProducts([]);
+    setValidateSuccessProducts([]);
```

Comment:
This is the main "fix" in this PR, the rest are just small refactors to make the code easier to follow.